src/rootless.rs

```diff
@@ -0,0 +1,121 @@
+use std::{env, path::PathBuf};
+
+use anyhow::{bail, Result};
+use nix::sched::CloneFlags;
+use oci_spec::{Linux, LinuxIdMapping, Mount, Spec};
+
+use crate::namespaces::Namespaces;
+
+#[derive(Debug, Clone)]
+pub struct Rootless {
+    /// Location of the newuidmap binary
+    pub newuidmap: Option<PathBuf>,
+    /// Location of the newgidmap binary
+    pub newgidmap: Option<PathBuf>,
+    /// Mappings for user ids
+    pub uid_mappings: Vec<LinuxIdMapping>,
+    /// Mappings for group ids
+    pub gid_mappings: Vec<LinuxIdMapping>,
+}
+
+impl From<&Linux> for Rootless {
+    fn from(linux: &Linux) -> Self {
+        Self {
+            newuidmap: None,
+            newgidmap: None,
+            uid_mappings: linux.uid_mappings.clone(),
+            gid_mappings: linux.gid_mappings.clone(),
+        }
+    }
+}
+
+/// Checks if rootless mode should be used
+pub fn should_use_rootless() -> Result<bool> {
+    if !nix::unistd::geteuid().is_root() {
+        return Ok(true);
+    }
+
+    if let Ok("true") = std::env::var("YOUKI_USE_ROOTLESS").as_deref() {
```

Comment:
Since rootless works for now, let's output a warn log that there is an unimplemented feature when this function is `true`.

Reply:
Isn't it also necessary to have `uidMapping` and `gidMapping` in the config?

Reply:
Yes, this is checked in the validate() function (L50-56)

Reply:
@Furisto Is it persistent to switch rootless with environment variables? Or is it temporary?

Reply:
I used it during development. I can remove it if you want.